src/io/parsers/GROParser.js

```diff
@@ -185,22 +182,19 @@ class GROParser extends Parser {
     /* Parse input file line-by-line */
     const reader = new GROReader(this._data);
     let counter = 0; /* Simple counter regarding to format of .gro file */
-    while (!reader.end()) {
-      /* First two lines - technical information, other lines - Atoms */
-      if (counter === 0) {
-        this._parseTitle(reader);
-      } else if (counter === 1) {
-        this._parseNumberOfAtoms(reader);
-      } else if (counter <= this._numAtoms + 1) { /* Number of atoms + 2 strings of technical information */
-        this._parseAtom(reader); /* Box Vectors are not been proceeded at all */
+    /* First two lines - technical information, other lines - Atoms */
+    this._parseTitle(reader);
+    reader.next();
+    this._parseNumberOfAtoms(reader);
+    reader.next();
+    for (counter = 0; counter < this._numAtoms; ++counter) {
+      if (!reader.end()) {
+        this._parseAtom(reader);
+        reader.next();
       } else break;
-      /* End of parsing, switch to next line */
-      reader.next();
-      counter++;
     }
-
     /* If number of atoms in second line is less then actual atoms in file */
-    if (counter < this._numAtoms + 1) {
+    if (counter < this._numAtoms) {
       this._complex.error = {
         message: 'File ended unexpectedly.',
```

Comment:
I'm not quite sure... Should we throw here too?

Reply:
There is some technical information in the last line of .gro file, so if counter < this._numAtoms then we 100% tried to parse that information as some kind of atom (which is not an atom), and that's a mistake.
I guess if we have mismatch in actual number of atoms and number that represents it - we have corrupted file, so therefore we must inform user about it? That's just an idea.